### PR TITLE
Libp2p Connection Limits

### LINKED ISSUE #798: Limit max active peers
**Issue summary**

Technically this can be a security vulnerability because a large amount of lightweight nodes which can strain on resources. 

This is definitely low priority for now, because it would require also correctly managing and disconnecting bad peers to make sure this doesn't become an actual dos vector.

**Other information and links**

### PR DESCRIPTION
**Pull request by ec2**
**Summary of changes**
Changes introduced in this pull request:
- Add new flag `--target-peer-count` to have a sort of bound on the number of peers we should be connected to with a default of 75. 
- Added Topic scoring to Gossipsub topics. They are currently disabled because I don't think were using them properly, but they aren't important for now, but might be in the future, so just leaving it in for posterity. 

**Reference issue to close (if applicable)**

Closes 
#798 

**Other information and links**

### CODE REVIEW

File: forest/src/cli/mod.rs [hunk 1]
```diff
@@ -128,6 +133,9 @@ impl DaemonOpts {
 
         cfg.network.kademlia = self.kademlia.unwrap_or(cfg.network.kademlia);
         cfg.network.mdns = self.mdns.unwrap_or(cfg.network.mdns);
+        if let Some(target_peer_count) = &self.target_peer_count {
+            cfg.network.target_peer_count = *target_peer_count;
+        }
```
Comment (cryptoquick) [inline forest/src/cli/mod.rs:138]: You don't need to reference and dereference here.

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Minor nitpick, otherwise, looks good.

File: forest/src/cli/mod.rs (hunk 1 above)
Comment (ec2) [reply forest/src/cli/mod.rs:138]: done

**Review by cryptoquick — APPROVED** [review]
Looks good.

**Review by creativcoder — APPROVED** [review]

**Pull request merged**